dev-util/dub/dub-1.38.1-r1.ebuild

```diff
@@ -0,0 +1,142 @@
+BDEPEND=${DLANG_DEPS}
+RDEPEND="${DLANG_DEPS}
+	virtual/pkgconfig
+"
+
+src_unpack() {
+	dub_gen_settings
+	unpack "${P}.tar.gz"
+
+	if use test; then
+		# Some tests do need to perform an actual fetch operation
+		local dep
+		for dep in "${MY_DUB_DEPENDENCIES[@]}"; do
+			cp "${DISTDIR}/${dep/@/-}.zip" "${DUB_LOCAL_REGISTRY}" || die
+		done
+	fi
+}
+
+src_configure() {
+	# gdc generates unaligned memory accesses with optimizations and avx
+	# enabled. It has been fixed upstream. See:
+	# https://gcc.gnu.org/bugzilla/show_bug.cgi?id=114171
+	# Fixed in >=sys-devel/gcc-13.2.1_p20240330. Adding -mno-sse2 makes
+	# tests fail so defer to removing the common way users get avx
+	# instructions enabled (-march=native) and warn them.
+	if [[ ${ARCH} == amd64 && ${EDC} == gdc-13 && ${DCFLAGS} == *-march=native* ]]; then
+		ewarn "<sys-devel/gcc-13.2.1_p20240330 is known to generate invalid code"
+		ewarn "on amd64 with certain flags. For this reason -march=native will be"
+		ewarn "removed from your flags. Feel free to use -march=<cpu> to bypass this"
+		ewarn "precaution."
+		ewarn ""
+		ewarn "See also: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=114171"
+		dlang-filter-dflags "gdc*" "-march=native"
+	fi
+}
+
+src_compile() {
+	local imports=source versions="DubApplication DubUseCurl"
+	dlang_compile_bin bin/dub $(<build-files.txt)
+
+	# Generate man pages. Rebuids dub so put it behind a USE flag.
+	if use doc; then
+		einfo "Generating man pages"
+		# You're supposed to be able to do ./bin/dub scrips/man/gen_man.d
+		# but it gives linking errors with gdc.
+
+		# $imports is set up above.
+		versions=DubUseCurl
+		dlang_compile_bin scripts/man/gen_man{,.d} \
+						  $(sed '/^source\/app.d$/d' build-files.txt)
+		./scripts/man/gen_man || die "Could not generate man pages"
+	fi
+}
+
+src_test() {
+	# Setup the environment for the tests.
+	local -x DUB="${S}/bin/dub"
+
+	# Note, disabling tests is possible yet very hard. You have to
+	# create a bash variable containing a regex (to be used in =~) that
+	# matches all the tests that you want *to* run. It's probably easier
+	# to delete the subdirectory under ${S}/test.
+
+	# Tries to connect to github.com and fails due to the network sandbox
+	rm -rf "${S}/test/git-dependency" || die
+	# Doesn't work on non amd64/x86
+	if [[ ${ARCH} == arm64 ]]; then
+		rm -rf test/issue1447-build-settings-vars || die
+	fi
+
+	# gdc-13 doesn't support #include's in its importC implementation.
+	if [[ ${EDC} == gdc-13 ]]; then
+		rm -rf "${S}/test/use-c-sources" || die
+	fi
+
+	# See https://bugs.gentoo.org/921581 we have to remove -op (preserve
+	# source path for output files) from the flags lest the sandbox
+	# trips us up. This shouldn't be a problem anymore with dlang-single.
+	dlang-filter-dflags "*" "--op" "-op"
+
+	# Use -Wno-error or equivalent
+	local -x DFLAGS="${DCFLAGS} ${DLANG_LDFLAGS} $(dlang_get_wno_error_flag)"
+
+	# Run the unittests in the source files.
+	"${DUB}" test --verbose -c application
```

Comment:
Missing die?

Reply:
Or was this supposed to be edub?

Reply:
No, it's missin the `die`. `edub` is not applicable as it will call the *assumed to be installed* dub, not the one which was just built. I don't think dub depending on dub is a good design so no calling `edub`.

Reply:
The pattern for this is to inhibit dependencies in the eclass if CAT/PN = blah/dub but it can be a bit ugly. Not required at all, just a suggestion.